Feat: Indexed connect + channel merger/splitter (#742)
software-mansion-labs/react-native-audio-api: packages/audiodocs/docs/core/audio-node.mdx

| Quad | 4 | 0: L - left <br /> 1: R - right <br /> 2: SL - surround left <br /> 3: SR - surround right |
| Stereo | 6 | 0: L - left <br /> 1: R - right <br /> 2: C - center <br /> 3: LFE - subwoofer <br /> 4: SL - surround left <br /> 5: SR - surround right |

#### Number of inputs and outputs

Each node has a specified `numberOfInputs` and `numberOfOutputs`. These represent multiple logical processing paths in the node, and currently are used only by the [`ChannelMergerNode`](/docs/effects/channel-merger-node) and [`ChannelSplitterNode`](/docs/effects/channel-splitter-node). 
The rest of the nodes have at most one input and output. You can connect multiple nodes to a single input, but keep in mind that the data will be put through a mixing process described below.

#### Mixing

> When a nodes input has more then one input or number of inputs channels differs from output up-mixing or down-mixing must be conducted.

Reviewer: ```suggestion
When a node has multiple inputs, or when the number of input and output channels  differs, up-mixing or down-mixing must be conducted.
```
